swim/memberlist.go

```diff
@@ -233,6 +233,7 @@ func (m *memberlist) RandomPingableMembers(n int, excluding map[string]bool) []*
 // returns an immutable slice of members representing the current state of the membership
 func (m *memberlist) GetMembers() (members []Member) {
 	m.members.RLock()
+	members = make([]Member, 0, len(m.members.list))
 	for _, member := range m.members.list {
```

Comment:
There is two reasons:

1. The two slices are of different types, the original slice is `[]*Member` which is a pointer to a struct, the returning slice is `[]Member` which puts the struct in place in the slice. This makes sure that the receiver of the slice is not able to change the members kept in memory of `memberlist`.
2. In the future a predicate will be passed into this function to return a filtered list of members eg. based on labels.